float8_experimental/float8_linear_utils.py

```diff
@@ -185,103 +185,110 @@ def sync_float8_amax_and_scale_history(model: torch.nn.Module, fp8_layers=None)
+        new_w_scales = amax_history_to_scale_stack(
+            fp8_w_amax_history_stack, torch.float8_e4m3fn, x_dtype, scale_fn_recipe
+        )
+        new_dL_dY_scales = amax_history_to_scale_stack(
+            fp8_dL_dY_amax_history_stack, torch.float8_e5m2, x_dtype, scale_fn_recipe
         )
-        if isinstance(all_reduced_amax_tensor, AsyncCollectiveTensor):
-            all_reduced_amax_tensor = all_reduced_amax_tensor.wait()
-
-        (
-            reduced_fp8_amax_tensor,
-            reduced_fp8_amax_w_tensor,
-            reduced_fp8_amax_dL_dY_tensor,
-        ) = torch.split(all_reduced_amax_tensor, len(fp8_amax_x_tensor_list))
 
+        # Iterate through the layers and update the scales, and set the flag to signal that the amaxes/scales are ready
         for idx, child in enumerate(fp8_layers):
-            child.fp8_amax_x.copy_(reduced_fp8_amax_tensor[idx])
-            child.fp8_amax_w.copy_(reduced_fp8_amax_w_tensor[idx])
-            child.fp8_amax_dL_dY.copy_(reduced_fp8_amax_dL_dY_tensor[idx])
-
-    # We create two stacked tensor groups, one for the amax history and one for the current scales
-    fp8_amax_x_tensors = torch.vstack(fp8_amax_x_tensor_list)
-    fp8_amax_w_tensors = torch.vstack(fp8_amax_w_tensor_list)
-    fp8_amax_dL_dY_tensors = torch.vstack(fp8_amax_dL_dY_tensor_list)
-
-    fp8_x_amax_history_stack = torch.vstack(fp8_x_amax_history_stack)
-    fp8_w_amax_history_stack = torch.vstack(fp8_w_amax_history_stack)
-    fp8_dL_dY_amax_history_stack = torch.vstack(fp8_dL_dY_amax_history_stack)
-
-    # Update the history stacks with the new amax values
-    _update_history_stack(fp8_amax_x_tensors, fp8_x_amax_history_stack)
-    _update_history_stack(fp8_amax_w_tensors, fp8_w_amax_history_stack)
-    _update_history_stack(fp8_amax_dL_dY_tensors, fp8_dL_dY_amax_history_stack)
-
-    # Calculate the new scales from the updated history stacks
-    new_x_scales = amax_history_to_scale_stack(
-        fp8_x_amax_history_stack, torch.float8_e4m3fn, x_dtype, scale_fn_recipe
-    )
-    new_w_scales = amax_history_to_scale_stack(
-        fp8_w_amax_history_stack, torch.float8_e4m3fn, x_dtype, scale_fn_recipe
-    )
-    new_dL_dY_scales = amax_history_to_scale_stack(
-        fp8_dL_dY_amax_history_stack, torch.float8_e5m2, x_dtype, scale_fn_recipe
-    )
+            child.fp8_scale_x.copy_(new_x_scales[idx])
+            child.fp8_scale_w.copy_(new_w_scales[idx])
+            child.fp8_scale_dL_dY.copy_(new_dL_dY_scales[idx])
 
-    # Iterate through the layers and update the scales, and set the flag to signal that the amaxes/scales are ready
-    for idx, child in enumerate(fp8_layers):
-        child.fp8_scale_x.copy_(new_x_scales[idx])
-        child.fp8_scale_w.copy_(new_w_scales[idx])
-        child.fp8_scale_dL_dY.copy_(new_dL_dY_scales[idx])
+    # This allows for the compile to succede on the inner func and fail on the graph breaks
+    # at the beginning and and of syncing
+    inner_func()
```

Comment:
a little hard to tell but I think this is all code motion? (moving it inside of inner_func())

Reply:
yeah tabs strike again here, but just indented one layer